Source/AssetRipper.Export.Modules.Shader/ShaderBlob/ShaderSubProgramBlob.cs

```diff
@@ -97,7 +97,7 @@ private static T[] CreateAndInitializeArray<T>(int length) where T : new()
 		}
 
 
-		public ShaderSubProgram GetSubProgram(uint blobIndex)
+		public ShaderSubProgram GetSubProgram(uint blobIndex, Action<ShaderSubProgram>? onRead)
```

Comment:
This feels like an antipattern.

Reply:
I know this is not a good way
Should CommonParameter be passed as a parameter, or set a out parameter to check if SubProgram is from cache, or just clone the ShaderSubProgram?

Reply:
It feels like you're trying to process these values after reading them. I think we need 2 object models:
* A low level layer that adheres very closely to the binary content.
* A high level layer that includes processing where necessary.